test/Feature/AtomicAdd.c

```diff
@@ -0,0 +1,14 @@
+// RUN: %llvmgcc %s -emit-llvm -g -c -o %t.bc
+// RUN: rm -rf %t.klee-out
+// RUN: %klee --output-dir=%t.klee-out --stop-after-n-instructions=500 --search=bfs %t.bc 2>%t.log
```

Comment:
Why `--stop-after-n-instructions=500 --search=bfs`? These should be unnecessary. 

Also `// RUN: rm -rf %t.klee-out` and `--output-dir=%t.klee-out` seem unnecessary.

I think it's also better to merge all these tests into 1 file (will also make it easier to change these flags).

Also older llvm versions seem to still fail. Maybe it's just easier to disable these tests for version less than 3.7. C++ is only 3.8+ anyway. What do you think @MartinNowack .

Reply:
@kren1 @futile Please use one single c test file. Each atomic instruction can be one function to be tested. And it would be nice to test semantics of that if the operation is correctly executed.

Beside the LLVM version, 3.4 already supports atomics. With a c file this should be easy.
To be sure that there are still the correct intrinsics you can easily `FileCheck` the `assembly.ll`.

If there is anything unclear, please feel free to ask.

Looking forward to see this getting merged ;)